src/repo/xdb_repo_queryable.erl

```diff
@@ -145,7 +145,7 @@ transaction(Repo, Adapter, Fun) ->
   Adapter :: xdb_adapter:t(),
   Fun     :: fun(() -> any()),
   Opts    :: xdb_lib:keyword(),
-  Res     :: {ok, any()} | {error, any()}.
+  Res     :: {ok, any()} | {error, any()} | {error, any(), [tuple()]}.
```

Comment:
@maximvl according to your thought `it's up to an adapter to implement this feature`, I think we shouldn't modify this spec, `{error, any()}` should be enough, you can pass the ST like `{error, {Reason, ST}}`, again, it is up to the adapter right?

Reply:
@cabol in a way yes, but the problem is the original error might look like `{Reason, Data}` and then it becomes hard to know for sure.